examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -70,34 +73,55 @@ public static void forward(BitcoinNetwork network, Address address) {
         System.out.println("Forwarding address: " + address);
 
         // Create the Service (and WalletKit)
-        ForwardingService forwardingService = new ForwardingService(address, network);
+        ForwardingService forwardingService = new ForwardingService(network);
 
         // Start the Service (and WalletKit)
         forwardingService.start();
 
         // Start listening and forwarding
-        forwardingService.waitForCoins();
+        CompletableFuture<Transaction> forwardedTxFuture = forwardingService.waitForCoins()
+            .thenCompose(tx -> {
+                // Incoming transaction received, now "compose" (i.e. chain) a call to wait for required confirmations
+                // The transaction "tx" can either be pending, or included into a block (we didn't see the broadcast).
+                Coin value = tx.getValueSentToMe(forwardingService.kit.wallet());
+                System.out.printf("Received tx for %s : %s\n", value.toFriendlyString(), tx);
+                System.out.println("Transaction will be forwarded after it confirms.");
+                System.out.println("Waiting for confirmation...");
+                return forwardingService.waitForConfirmation(tx);
+            })
+            .thenCompose(confidence -> {
+                // Required confirmations received, now compose a call to broadcast the forwarding transaction
+                System.out.printf("Incoming tx has received %d confirmations.", confidence.getDepthInBlocks());
+                // Now send the coins onwards by sending the entire contents of our wallet
+                SendRequest sendRequest = SendRequest.emptyWallet(address);
+                try {
+                    forwardingService.kit.wallet().completeTx(sendRequest);
+                    System.out.printf("Sending tx %s ...\n", sendRequest.tx.getTxId());
```

Comment:
This code might be little bit cleaner if we factored out a `buildForwardingTransaction()` (or `buildRefundTransaction()` method that builds and returns a `Transaction`